kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/IndexNamingStrategy.scala

```diff
@@ -0,0 +1,32 @@
+package io.lenses.streamreactor.connect.aws.s3.sink
+
+import io.lenses.streamreactor.connect.aws.s3.model.{BucketAndPath, BucketAndPrefix, Offset, Topic, TopicPartition, TopicPartitionOffset}
+
+import scala.util.matching.Regex
+
+class S3IndexNamingStrategy {
+
+  protected val DefaultPrefix = "index"
+
+  def prefix(bucketAndPrefix: BucketAndPrefix): String = bucketAndPrefix.prefix.getOrElse(DefaultPrefix)
+
+  def indexPath(bucketAndPrefix: BucketAndPrefix, topicPartition: TopicPartition): BucketAndPath =
+    BucketAndPath(bucketAndPrefix.bucket, s"${prefix(bucketAndPrefix)}/${topicPartition.topic.value}/${topicPartition.partition}/")
+
+  def indexFilename(bucketAndPrefix: BucketAndPrefix, topicPartitionOffset: TopicPartitionOffset): BucketAndPath =
```

Comment:
wouldn't this be putting files in the same target bucket with the fileswith the actual data? And if that's the case listing the files will be costly since the list will grow with times